kartograf/merge.py

```diff
@@ -139,20 +164,37 @@ def general_merge(
     Merge lists of IP networks into a base file.
     """
     print("Merging extra prefixes that were not included in the base file.")
-    base = BaseNetworkIndex()
+    base_network_index = BaseNetworkIndex()
     with open(base_file, "r") as file:
         for line in file:
             pfx, _ = line.split(" ")
-            base.update(pfx)
+            base_network_index.update(pfx)
 
     df_extra = extra_file_to_df(extra_file)
 
-    extra_included = []
-    for row in df_extra.itertuples(index=False):
-        result = base.contains_row(row)
-        extra_included.append(result)
+    chunk_size = 50000
+    chunks = []
+    for i in range(0, len(df_extra), chunk_size):
+        chunk = df_extra.iloc[i:i+chunk_size]
+        # get each chunk with an index
+        chunk_data = [
+            (i+j, (row.INETS, row.ASNS, row.PFXS, row.PFXS_LEADING))
+            for j, row in enumerate(chunk.itertuples(index=False))
+        ]
+        chunks.append(chunk_data)
+
+    with ProcessPoolExecutor() as executor:
+        base_dict = base_network_index.get_serializable_dict()
```

Comment:
Is there a reason why the `base_dict` is handed around and not and instance of `BaseNetworkIndex`? I think it would be cleaner unless there there is some reason that prevents that.

Reply:
yea, the _dict.keys() thing is not picklable, and everything passed to ProcessPollExecutor must be picklable.